### PR TITLE
modulestream C test, builds atop PR #118

### PR DESCRIPTION
**Pull request by asamalik**
Adding a test case for modulestream in C. 

(My first C code in years!)

### CODE REVIEW

File: modulemd/v2/tests/test-modulemd-modulestream.c [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      g_clear_object (&copied_stream);
+    }
+}
+
+int
+main (int argc, char *argv[])
+{
+  setlocale (LC_ALL, "");
+  g_test_init (&argc, &argv, NULL);
+  g_test_bug_base ("https://bugzilla.redhat.com/show_bug.cgi?id=");
```
Comment (ignatenkobrain) [inline modulemd/v2/tests/test-modulemd-modulestream.c]: nope ;)

**Comment by asamalik** [comment]
I've done it horribly wrong, let me rebase it on top of the PR #118!

File: meson_options.txt [hunk 2]
```diff
@@ -11,6 +11,6 @@
 
 option('developer_build', type : 'boolean', value : true)
 option('build_api_v1', type : 'boolean', value : true)
-option('build_api_v2', type : 'boolean', value : false)
+option('build_api_v2', type : 'boolean', value : true)
```
Comment (sgallagher) [inline meson_options.txt]: Please don't change the default options. That will happen once 2.0 is released. Right now, you should just specify `meson -Dbuild_api_v2=true <dir>` when configuring the project.

File: modulemd/v2/tests/test-modulemd-modulestream.c (hunk 1 above)
Comment (sgallagher) [reply modulemd/v2/tests/test-modulemd-modulestream.c]: Could you leave a more productive comment, @ignatenkobrain ? What part do you disagree with?

File: modulemd/v2/tests/test-modulemd-modulestream.c [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+      g_clear_object (&stream);
+    }
+
+  /* Test with a zero mdversion */
+  stream = modulemd_module_stream_new (0, "foo", "latest");
+  g_assert_null (stream);
+  g_clear_object (&stream);
+
+
+  /* Test with a zero mdversion */
```
Comment (sgallagher) [inline modulemd/v2/tests/test-modulemd-modulestream.c]: Comment doesn't match the test.

**Comment by sgallagher** [comment]
@asamalik Also, please squash both of your patches together.

File: meson_options.txt (hunk 2 above)
Comment (asamalik) [reply meson_options.txt]: Thanks for the tip, changing it back.

**Comment by asamalik** [comment]
@sgallagher fixed!

**Pull request merged**